models/convert/cadence.go

```diff
@@ -178,3 +180,48 @@ func ParseCadenceArgument(param string) (cadence.Value, error) {
 		return nil, fmt.Errorf("unknown type for Cadence conversion (%s)", typ)
 	}
 }
+
+// ParseRosettaValue parses the input string, which should be the Rosetta value, and
+// produces the Cadence UFix64 number. The function will assume that the input string
+// has dps.FlowDecimals decimal places.
+func ParseRosettaValue(value string) (cadence.UFix64, error) {
+
+	var decimal int64
+	var fraction uint64
+	var err error
+
+	// If the number is longer than dps.FlowDecimals (8) characters, it means
+	// that we will have both the decimal and the fractional part.
+	// The last 8 characters will represent the fraction, while anything before
+	// that is the decimal part.
+	// If the number is shorter, it means that the whole string is the fraction.
+	if len(value) > dps.FlowDecimals {
+
+		// Get all characters except the last 8 and convert that string to an integer.
+		decimal, err = strconv.ParseInt(value[:len(value)-dps.FlowDecimals], 10, 32)
+		if err != nil {
+			return 0, fmt.Errorf("could not parse decimal part: %w", err)
+		}
+
+		// Get the last 8 characters and convert them to an uint.
+		fraction, err = strconv.ParseUint(value[len(value)-dps.FlowDecimals:], 10, 32)
```

Comment:
```suggestion
		decimalsIndex := len(value)-dps.FlowDecimals

		// Get all characters except the last 8 and convert that string to an integer.
		decimal, err = strconv.ParseInt(value[:decimalsIndex ], 10, 32)
		if err != nil {
			return 0, fmt.Errorf("could not parse decimal part: %w", err)
		}

		// Get the last 8 characters and convert them to an uint.
		fraction, err = strconv.ParseUint(value[decimalsIndex :], 10, 32)
```

WDYT? Makes it slightly simpler IMO :)

Reply:
Yeah, I agree :) Will change